src/components/ProgressBar.tsx

```diff
@@ -82,18 +82,20 @@ export function ProgressBar({
   };
 
   if (controlledProgress) style.transform = `scaleX(${progress})`;
-
-  const classNames = cx(
+  const defaultClassArr = [
     `${DEFAULT.CSS_NAMESPACE}__progress-bar`,
     controlledProgress
       ? `${DEFAULT.CSS_NAMESPACE}__progress-bar--controlled`
       : `${DEFAULT.CSS_NAMESPACE}__progress-bar--animated`,
     `${DEFAULT.CSS_NAMESPACE}__progress-bar--${type}`,
     {
       [`${DEFAULT.CSS_NAMESPACE}__progress-bar--rtl`]: rtl
-    },
-    className
-  );
+    }
+  ];
+  const classNames =
+    typeof className === 'function'
```

Comment:
May I ask to replace `typeof className === 'function'` by isFn(className)? Minify better

Reply:
Done